MSMAll/scripts/MSMAll.sh

```diff
@@ -1,1195 +1,254 @@
+opts_AddOptional '--matlab-run-mode' 'MatlabRunMode' '0, 1, or 2' "defaults to $g_matlab_default_mode
+0 = compiled MATLAB
+1 = interpreted MATLAB
+2 = Octave" "$g_matlab_default_mode"
+opts_ParseArguments "$@"
+
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
 
-# ------------------------------------------------------------------------------
-#  Show usage information for this script
-# ------------------------------------------------------------------------------
+# Verify required environment variables are set and log value
+log_Check_Env_Var HCPPIPEDIR
+log_Check_Env_Var CARET7DIR
+log_Check_Env_Var MSMBINDIR
+
+#display HCP Pipeline version
+log_Msg "Showing HCP Pipelines version"
+"${HCPPIPEDIR}"/show_version --short
 
-show_usage()
-{
-	cat <<EOF
-
-${g_script_name}: MSM-All Registration
-
-Usage: ${g_script_name} PARAMETER...
-
-PARAMETERs are [ ] = optional; < > = user supplied value
-
-  TBW = To Be Written
-
-  [--help] : show usage information and exit
-   --path=<path to study folder> OR --study-folder=<path to study folder>
-   --subject=<subject ID>
-   --high-res-mesh=<high resolution mesh node count> (in thousands)
-   --low-res-mesh=<low resolution mesh node count> (in thousands)
-   --output-fmri-name=<name given to concatenated singel subject "scan">
-   --fmri-proc-string=<identification for FIX cleaned dtseries to use>
-   --input-pca-registration-name=TBW
-   --input-registration-name=TBW
-   --registration-name-stem=TBW
-   --rsn-target-file=TBW
-   --rsn-cost-weights=TBW
-   --myelin-target-file=TBW
-   --topography-roi-file=TBW
-   --topography-target-file=TBW
-   --iterations=TBW
-   --method=TBW
-   --use-migp=TBW
-   --ica-dim=TBW
-   --regression-params=TBW
-   --vn=TBW
-   --rerun=TBW
-   --reg-conf=TBW
-   --reg-conf-vars=TBW
-  [--matlab-run-mode={0, 1, 2}] defaults to ${G_DEFAULT_MATLAB_RUN_MODE}
-     0 = Use compiled MATLAB
-     1 = Use interpreted MATLAB
-     2 = Use Octave
-
-EOF
-}
+#display the parsed/default values
+opts_ShowValues
 
 # ------------------------------------------------------------------------------
-#  Get the command line options for this script.
+#  Main processing of script.
 # ------------------------------------------------------------------------------
+# Naming Conventions and other variables
+Caret7_Command=${CARET7DIR}/wb_command
+AtlasFolder="${StudyFolder}/${Subject}/MNINonLinear"
+DownSampleFolder="${AtlasFolder}/fsaverage_LR${LowResMesh}k"
+NativeFolder="${AtlasFolder}/Native"
+ResultsFolder="${AtlasFolder}/Results/${OutputfMRIName}"
+T1wFolder="${StudyFolder}/${Subject}/T1w"
+DownSampleT1wFolder="${T1wFolder}/fsaverage_LR${LowResMesh}k"
+NativeT1wFolder="${T1wFolder}/Native"
+
+if [[ $(echo -n "${Method}" | grep "WR") ]] ; then
+	LowICAdims=$(echo "${RegressionParams}" | sed 's/@/ /g')
```

Comment:
I think Matt wanted to still support `_` here.  You could do this as-is with sed using `'s/[_@]/ /g'`.  There is also a way to do it with arrays, but since these will all be numbers, we don't need to worry about handling whitespace.